virttest/test_setup.py

```diff
@@ -487,18 +487,20 @@ def set_hugepages(self):
         error_context.context(
             "setting hugepages limit to %s" % self.target_hugepages)
         hugepage_cfg = open(self.kernel_hp_file, "r+")
-        hp = hugepage_cfg.readline()
+        hp = hugepage_cfg.readline().strip()
+        hugepage_cfg.close()
         while int(hp) < self.target_hugepages:
             loop_hp = hp
+            hugepage_cfg = open(self.kernel_hp_file, "r+")
```

Comment:
it is better to use try and except with IOError and raise exceptions.TestError()